common/test/acceptance/pages/lms/pay_and_verify.py

```diff
@@ -0,0 +1,159 @@
+    a deferred verification, or when attempting to view a receipt corresponding
+    to an earlier payment.
+    """
+    def __init__(self, browser, course_id, entry_point='start-flow'):
+        """Initialize the page.
+
+        Arguments:
+            browser (Browser): The browser instance.
+            course_id (unicode): The course in which the user is enrolling.
+
+        Keyword Arguments:
+            entry_point (str): Where to begin the flow; must be one of 'start-flow',
+                'upgrade', 'verify-now', verify-later', or 'payment-confirmation'.
+
+        Raises:
+            ValueError
+        """
+        super(PaymentAndVerificationFlow, self).__init__(browser)
+        self._course_id = course_id
+
+        if entry_point not in ['start-flow', 'upgrade', 'verify-now', 'verify-later', 'payment-confirmation']:
+            raise ValueError(
+                "Entry point must be either 'start-flow', 'upgrade', 'verify-now', 'verify-later', or 'payment-confirmation'."
+            )
+        self._entry_point = entry_point
+
+    @property
+    def url(self):
+        """Return the URL corresponding to the initial position in the flow."""
+        url = "{base}/verify_student/{entry_point}/{course}".format(
+            base=BASE_URL,
+            entry_point=self._entry_point,
+            course=self._course_id
+        )
+
+        return url
+
+    def is_browser_on_page(self):
+        """Check if a step in the payment and verification flow has loaded."""
+        return (
+            self.q(css="div .make-payment-step").is_present() or
+            self.q(css="div .payment-confirmation-step").is_present() or
+            self.q(css="div .face-photo-step").is_present() or
+            self.q(css="div .id-photo-step").is_present() or
+            self.q(css="div .review-photos-step").is_present() or
+            self.q(css="div .enrollment-confirmation-step").is_present()
+        )
+
+    def indicate_contribution(self):
+        """Interact with the radio buttons appearing on the first page of the upgrade flow."""
+        self.q(css=".contribution-option > input").first.click()
+
+    def proceed_to_payment(self):
+        """Interact with the payment button."""
+        self.q(css="#pay_button").click()
+
+        FakePaymentPage(self.browser, self._course_id).wait_for_page()
+
+    def immediate_verification(self):
+        """Interact with the immediate verification button."""
+        self.q(css="#verify_now_button").click()
+
+        PaymentAndVerificationFlow(self.browser, self._course_id, entry_point='verify-now').wait_for_page()
+
+    def defer_verification(self):
+        """Interact with the link allowing the user to defer their verification."""
+        self.q(css="#verify_later_button").click()
+
+        DashboardPage(self.browser).wait_for_page()
+
+    def webcam_capture(self):
+        """Interact with a webcam capture button."""
+        self.q(css="#webcam_capture_button").click()
```

Comment:
It might be worth adding a check here that the "#next_step_button" is enabled before returning control to the caller.